packages/client/lib/net/server/rlpxserver.ts

```diff
@@ -51,6 +52,21 @@ export class RlpxServer extends Server {
   constructor(options: RlpxServerOptions) {
     super(options)
 
+    if (this.key === undefined) {
```

Comment:
nice, looks good, can you put this logic in a helper func in `config.ts` named something like `getClientKey(): Buffer` that does the dual get/save logic? Then I can reuse in `libp2pserver.ts` in my pr.

And then here it can just be `this.key = getClientKey()`

Reply:
Yep - will do! Do you have any suggestion to fix the browser tests not passing? It doesn't have access to this `fs-extra` stuff, how do we fix this problem in general? We have a lot of `fs`-related operations right?

Reply:
hmmm yeah browser won't have access to fs-related modules.

In the client readme it says:

> That's it! Now, you should start seeing headers being downloaded to the local storage of your browser. Since IndexDB is being used, even if you close and re-open the browser window, the headers you've downloaded will be saved.

Maybe we can do something similarly like if we are in the browser we can save the key to localStorage and get it from there as well.

I wonder if there is a package that can abstract away this `fs || localStorage` logic with a higher level api. (briefly googling around, found https://github.com/jvilk/BrowserFS although haven't looked further if it would be a good choice for our usage here)

Reply:
Can we maybe just use leveldb for this as well? Is another separate DB instance for configuration too heavy?